decidim-core/app/models/decidim/user.rb

```diff
@@ -113,6 +115,14 @@ def self.has_pending_invitations?(organization_id, email)
       invitation_not_accepted.find_by(decidim_organization_id: organization_id, email:)
     end
 
+    def marked_for_deletion?
+      marked_for_deletion_at.present?
+    end
+
+    def removable?(removal_period)
+      marked_for_deletion? && last_sign_in_at < Time.current - removal_period
+    end
```

Comment:
```suggestion
    def removable?
      last_sign_in_at < Decidim.delete_inactive_users_after_days.ago
    end
```
Althoug it might not be needed anymore